lib/honeybadger/agent.rb

```diff
@@ -384,6 +388,12 @@ def init_worker
       @worker = Worker.new(config)
     end
 
+    def with_error_handling
+      yield
+    rescue => ex
```

Comment:
I'm not sure that this is how we want to do the rescuing. What do you think @minhajuddin?

Reply:
This looks good, we should be rescuing `StandardError`s :+1: